resources/hab_build.rb

```diff
@@ -20,20 +22,28 @@
     command "sudo -E #{hab_binary} studio" \
             " -r #{hab_studio_path}" \
             " build #{plan_dir}"
-    environment('TERM' => 'vt100',
-                'HAB_ORIGIN' => origin,
-                'HAB_NONINTERACTIVE' => 'true')
+    environment({ 'TERM' => 'vt100',
+                  'HAB_ORIGIN' => origin,
+                  'HAB_NONINTERACTIVE' => 'true' }.merge(
+                    new_resource.environment
```

Comment:
@schisamo This is why rails invented `reverse_merge` ;) http://api.rubyonrails.org/classes/Hash.html#method-i-reverse_merge